### ISSUE TITLE
usefulness

### ISSUE THREAD

**Issue by Offirmo** [dphilipson/typescript-string-enums#1]
I fail to see the need for your lib.

Why not just using `type Status = "RUNNING" | "STOPPED"` ?

It's as simple and as type safe.

The only usage of traditional Enum I can see is for mapping to an int, which is not your target here.

**Comment by dphilipson** [comment]
Using string literals throughout your program opens you up to bugs caused by typos or incomplete refactors. An example:
``` javascript
type Status = "RUNNING" | "STOPPED";

...

// Typo "RUNING" is not caught by typechecker.
if (state.status === "RUNING") {
    ...
}
```
By contrast:
``` javascript
const Status = Enum("RUNNING", "STOPPED");
type Status = Enum<typeof Status>;

...

// Type error: property "RUNING" not found.
if (state.status === Status.RUNING) {
    ...
}
```

Furthermore, string literals make refactoring difficult. For instance, suppose I have two different enums:
``` javascript
type Status = "RUNNING" | "STOPPED";
type TransportationMode = "DRIVING" | "FLYING" | "RUNNING";
```
Then if I later decide I want to change `Status` to be `"STARTED" | "STOPPED"` instead, I don't have an easy way to do it. I can't globally find/replace "RUNNING" to "STARTED" because it will also change the unrelated string constants representing `TransportationMode`, so I have to examine every occurrance of the string "RUNNING" to see if it needs to change. With string enums, this is easy- I can do a global find/replace of "Status.RUNNING" to "Status.STOPPED", or alternatively change the enum definition and then visit each type error to fix it.

**Comment by dphilipson** [comment]
I appreciate your asking this question by the way- it was probably on many people's minds. I will add this to the readme when I have a chance.

**Comment by Offirmo** [comment]
Very interesting. I didn't know about those downsides.

**Comment by Offirmo** [comment]
Done, had an exact use case and added it to my project at work. (in PR)

**Comment by dphilipson** [comment]
Glad to hear! I hope it works out for you.

I just added the above explanations to the README, so I think now's a reasonable time to close this. Thank you again for reaching out!

**Comment by Offirmo** [comment]
Explanations are perfect 👌

**Comment by abenhamdine** [comment]
@dphilipson you should absolutely add this in the readme, I had the same question until I read this topic.

**Comment by dphilipson** [comment]
@abenhamdine It is in the readme in the "Motivation" section. Or do you find that explanation insufficient?

If you missed that section, maybe I should add a table of contents to the readme to make it more obvious.

**Comment by abenhamdine** [comment]
oups, indeed I missed it 😞 
Yes a table of contents should be nice, or simply move it at top of readme.

**Comment by dphilipson** [comment]
I've added a table of contents. I'd prefer not to put the Motivation section at the top because it's somewhat lengthy and I'd like readers to get to the code examples quickly.

Thank you for your feedback!
